### DISCUSSION TITLE
ANR (Application Not Responding) Issue Occurs When Executing JavaScript Function in a For Loop (array size of 5000)

### DISCUSSION THREAD

**Discussion by roshni-warrantylife** [mozilla/rhino#1500] [Q&A]
When attempting to execute a JavaScript function within a for loop in the Android application, an ANR (Application Not Responding) error occurs. The issue arises specifically during the execution of JavaScript functions in a loop, leading to unresponsiveness in the application's UI.

Version: 1.7.14

Logs: 
`main (runnable)
tid=1 systid=13810
Triggered ANR
Root blocking
org.mozilla.javascript.ScriptableObject.get (ScriptableObject.java:237)
org.mozilla.javascript.IdScriptableObject.get (IdScriptableObject.java:367)
org.mozilla.javascript.ScriptableObject.getProperty (ScriptableObject.java:2023)
org.mozilla.javascript.ScriptRuntime.getPropFunctionAndThisHelper (ScriptRuntime.java:2577)
org.mozilla.javascript.ScriptRuntime.getPropFunctionAndThis (ScriptRuntime.java:2568)
org.mozilla.javascript.Interpreter.interpretLoop (Interpreter.java:1658)
org.mozilla.javascript.Interpreter.interpret (Interpreter.java:1053)
org.mozilla.javascript.InterpretedFunction.call (InterpretedFunction.java:90)
org.mozilla.javascript.ContextFactory.doTopCall (ContextFactory.java:380)
org.mozilla.javascript.ScriptRuntime.doTopCall (ScriptRuntime.java:3868)
org.mozilla.javascript.InterpretedFunction.call (InterpretedFunction.java:88)`

I'm uncertain about the solution, but I'm hopeful that you might have some insights. Please let me know if you need further information or assistance in resolving this issue. Thank you!

**Comment by p-bakker** [comment]
What kind of 'solution' are you looking for? 

You're running some code that takes a long time to execute and apparently Android detects that and throws an error. 

So what are you going for posting a question here?
